assets/prior_versions_index.sql

```diff
@@ -0,0 +1,77 @@
+-- Step 1: Declare variables
+DECLARE idc_versions ARRAY<INT64>;
+DECLARE latest_idc_version INT64;
+DECLARE union_all_query STRING;
+
+-- Step 2: Get all idc_versions
+SET idc_versions = (
+  SELECT [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18]
+  --SELECT ARRAY_AGG(idc_version)
+  --FROM
+  --`bigquery-public-data.idc_current.version_metadata`
+);
+
+SET latest_idc_version = (
+  SELECT 18
+  --SELECT max(idc_version)
+  --FROM
+  --`bigquery-public-data.idc_current.version_metadata`
+);
+
+-- Step 3: Generate the UNION ALL query dynamically
+SET union_all_query = (
+  SELECT STRING_AGG(
+    FORMAT("""
+    SELECT
+    %d AS idc_version,
+    collection_id,
+    PatientID,
+    SeriesInstanceUID,
+    StudyInstanceUID,
+    Modality,
+    regexp_extract(gcs_url, 'gs://([^/]+)/') as gcs_bucket,
+    crdc_series_uuid,
+    ROUND(SUM(SAFE_CAST(instance_size AS float64))/1000000, 2) AS series_size_MB,
+  FROM
+  `bigquery-public-data.idc_v%d.dicom_all` AS dicom_all
+  where crdc_series_uuid not in (select distinct crdc_series_uuid from `bigquery-public-data.idc_v%d.dicom_all`)
+  GROUP BY
+  1,2,3,4,5,6,7,8
+
+""", version, version, latest_idc_version),
+    " UNION ALL "
+  )
+  FROM UNNEST(idc_versions) AS version
+);
+
+-- Step 4: Execute the complete query
+EXECUTE IMMEDIATE FORMAT("""
+WITH all_versions AS (
+  %s
+)
+SELECT
+  collection_id,
+  PatientID,
+  SeriesInstanceUID,
+  StudyInstanceUID,
+  Modality,
+  gcs_bucket,
+  crdc_series_uuid,
+  series_size_MB,
+  CASE
+  WHEN gcs_bucket='public-datasets-idc' THEN CONCAT('s3://','idc-open-data/',crdc_series_uuid, '/*')
```

Comment:
```suggestion
  # map GCS bucket to AWS bucket, since for idc-index we prefer AWS
  # if new buckets are included in IDC, this will need to be updated!
  WHEN gcs_bucket='public-datasets-idc' THEN CONCAT('s3://','idc-open-data/',crdc_series_uuid, '/*')
```